### DISCUSSION TITLE
Initial support for 'customizable' themes on 480x320 LCD panels

### DISCUSSION THREAD

**Discussion by tjko** [tjko/fanpico#30] [Show and tell]
Initial support for 480x320 size LCD panels is working.

Plan is to have support for "themes" where background image is displayed on the screen and then output 'fields' are rendered on locations specified by the theme with the colors (and font size) specified in the theme.

Initial test/default theme is now working:
![480x320theme](https://user-images.githubusercontent.com/1075726/226509071-4203e619-47a2-402b-9c44-5b96f8cda1e7.jpg)

**Comment by dmgeurts** [comment]
What kind of screen is this?

**Reply by tjko** [reply]
Screen in the photo is a Waveshare "3.5inch RPi LCD (B)" module meant for Raspberry Pi. It is pretty nice inexpensive IPS panel. (Much better than the "(A)" version that is few dollars cheaper but TFT with bad viewing angles...)

I added Wiki page to document panels that have been "tested":

https://github.com/tjko/fanpico/wiki/LCD-(SPI)-Panel-Connection

**Reply by dmgeurts** [reply]
Great work, thank you! I see that several of these screens are touch screens, I'm assuming that this is not supported by fanpico?

**Reply by tjko** [reply]
Yes, no support for touchscreen at the moment.  

On current boards all available pins on Pico are in already in use, so it probably would require a new PCB design that frees some pins on the Pico....

**Comment by tjko** [comment]
Support for custom themes should now be working. Hoping to get themes contributed to include in FanPico firmware.

https://github.com/tjko/fanpico/wiki/Creating-Custom-(LCD)-Themes
